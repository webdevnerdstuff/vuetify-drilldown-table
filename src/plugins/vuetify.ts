import '@mdi/font/css/materialdesignicons.css';
import 'vuetify/styles';
import defaultThemes from './theme';
import * as components from 'vuetify/components';
import * as directives from 'vuetify/directives';
import { aliases, mdi } from 'vuetify/iconsets/mdi';
import { createVuetify } from 'vuetify';

// ! This is Labs workaround, need to update later //
import * as labs from 'vuetify/labs/components';


export default createVuetify({
	components: {
		components,
		...labs,
	},
	directives,
	icons: {
		defaultSet: 'mdi',
		aliases,
		sets: {
			mdi,
		}
	},
	theme: {
		defaultTheme: 'light',
		themes: {
			...defaultThemes,
		},
	},
});
