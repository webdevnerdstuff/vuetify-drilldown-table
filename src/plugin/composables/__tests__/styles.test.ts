import { describe, it, expect } from 'vitest';
import {
	useTableStyles,
	useHeaderCellStyles,
	useTFootCellStyles,
} from '../styles';
import vuetify from '../../../plugins/vuetify';

const theme = vuetify.theme;

const colors = {
	default: {
		background: 'primary',
		border: 'primary',
		color: 'on-primary'
	},
	footer: {
		background: 'primary',
		color: 'on-primary'
	},
	header: {
		background: 'primary',
		color: 'on-primary'
	},
	percentageChange: 15,
	percentageDirection: 'desc'
};

describe('Styles Composable', () => {
	describe('useTableStyles', () => {
		it('should return tables styles for level 1', () => {
			const data = useTableStyles({
				colors,
				level: 1,
				theme,
			});
			expect(data).toMatchSnapshot();
		});

		it('should return tables styles for level 2', () => {
			const data = useTableStyles({
				colors,
				level: 2,
				theme,
			});
			expect(data).toMatchSnapshot();
		});
	});

	describe('useHeaderCellStyles', () => {
		const options = {
			colors,
			column: {
				width: undefined as string | number | undefined,
			},
			dataTableExpand: false,
			level: 1,
			theme,
		};

		it('should return header cell styles for level 1 using an undefined width to equal "auto"', () => {
			const data = useHeaderCellStyles(options);
			expect(data).toMatchSnapshot();
		});


		it('should return header cell styles for level 1 using an width as number', () => {
			options.column.width = 100;
			const data = useHeaderCellStyles(options);
			expect(data).toMatchSnapshot();
		});


		it('should return header cell styles for level 1 using an width as string without unit', () => {
			options.column.width = '100';
			const data = useHeaderCellStyles(options);
			expect(data).toMatchSnapshot();
		});

		it('should return header cell styles for level 1 using an width as string with unit', () => {
			options.column.width = '100px';
			const data = useHeaderCellStyles(options);
			expect(data).toMatchSnapshot();
		});

		it('should return header cell styles for level 2', () => {
			options.column.width = '100px';
			options.level = 2;
			const data = useHeaderCellStyles(options);
			expect(data).toMatchSnapshot();
		});
	});

	describe('useTFootCellStyles', () => {
		const options = {
			colors,
			elm: 'footer',
			level: 1,
			theme,
		};

		it('should return footer cell styles level 1', () => {
			const data = useTFootCellStyles(options);
			expect(data).toMatchSnapshot();
		});

		it('should return footer cell styles level 2', () => {
			options.level = 2;
			const data = useTFootCellStyles(options);
			expect(data).toMatchSnapshot();
		});
	});
});
