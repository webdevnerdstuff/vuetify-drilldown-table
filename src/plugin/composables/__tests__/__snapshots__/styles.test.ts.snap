// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Styles Composable > useHeaderCellStyles > should return header cell styles for level 1 using an undefined width to equal "auto" 1`] = `
{
  "backgroundColor": "hsl(207 90% 54% / 100%)",
  "color": "hsl(0 0% 100% / 100%)",
  "minWidth": "auto",
  "width": "auto",
}
`;

exports[`Styles Composable > useHeaderCellStyles > should return header cell styles for level 1 using an width as number 1`] = `
{
  "backgroundColor": "hsl(207 90% 54% / 100%)",
  "color": "hsl(0 0% 100% / 100%)",
  "minWidth": "100px",
  "width": "100px",
}
`;

exports[`Styles Composable > useHeaderCellStyles > should return header cell styles for level 1 using an width as string with unit 1`] = `
{
  "backgroundColor": "hsl(207 90% 54% / 100%)",
  "color": "hsl(0 0% 100% / 100%)",
  "minWidth": "100px",
  "width": "100px",
}
`;

exports[`Styles Composable > useHeaderCellStyles > should return header cell styles for level 1 using an width as string without unit 1`] = `
{
  "backgroundColor": "hsl(207 90% 54% / 100%)",
  "color": "hsl(0 0% 100% / 100%)",
  "minWidth": "100px",
  "width": "100px",
}
`;

exports[`Styles Composable > useHeaderCellStyles > should return header cell styles for level 2 1`] = `
{
  "backgroundColor": "hsl(207 90% 54% / 85%)",
  "color": "hsl(0 0% 100% / 100%)",
  "minWidth": "100px",
  "width": "100px",
}
`;

exports[`Styles Composable > useTFootCellStyles > should return footer cell styles level 1 1`] = `
{
  "backgroundColor": "hsl(207 90% 54% / 100%)",
  "color": "hsl(0 0% 100% / 100%)",
}
`;

exports[`Styles Composable > useTFootCellStyles > should return footer cell styles level 2 1`] = `
{
  "backgroundColor": "hsl(207 90% 54% / 85%)",
  "color": "hsl(0 0% 100% / 100%)",
}
`;

exports[`Styles Composable > useTableStyles > should return tables styles for level 1 1`] = `
{
  "borderBottom": "1px solid hsl(207 90% 54% / 100%)",
}
`;

exports[`Styles Composable > useTableStyles > should return tables styles for level 2 1`] = `
{
  "borderBottom": "1px solid hsl(207 90% 54% / 85%)",
}
`;
